### ISSUE TITLE
Need a diagnostic build inside VS that is actually diagnostic

### ISSUE THREAD

**Issue by davkean** [Microsoft/msbuild#1711]
Diagnostic builds inside VS used to contain property and items values at the start of the log. It was turned off for reasons that I cannot remember (probably perf).

When debugging builds inside VS, usually I can get away with falling back to command-line and using it's diagnostic mode (which does include property/items values). However, when debugging design-time issues - that's much harder, because there's a lot more influence at play such as [global properties](https://github.com/Microsoft/msbuild/issues/1710) and the targets that are being called.

I'd like either a option or a new mode that actually includes property and items values inside VS logging.

**Comment by rainersigwald** [comment]
Isn't this a property of the logger that VS (CPS?) is attaching? Properties and Items are available in MSBuild's `ProjectStartedEventArgs`, so it should be easy to log them.

**Comment by davkean** [comment]
Okay, after spending a non-trivial amount of time walking through MSBuild's logging infrastructure (oh my is there a lot of indirection!) - I found out the cause and the right place to change this.

The default logger that we use (ConsoleLogger), already logs parameters, items and tasks inputs if they are specified. However, they weren't getting passed through to it. Turns out, it was because this was turned off via BuildParameters and it was actually only turned for Design-Time builds.

The solution build manager has the following:

```
                if (verbosity == LoggerVerbosity.Diagnostic)
                {
                    buildParameters.LogTaskInputs = true;
                    buildParameters.LogInitialPropertiesAndItems = true;
                }
```

The CPS design-time build manager doesn't turn this on. The fix is to do that.
